### PR TITLE
[REVIEW] Fix NVStrings test_convert failure in 10.2 build

### PR DESCRIPTION
**Pull request by davidwendt**
The `test_convert` module was using `EXPECT_EQ` for checking float and double values.
Changed the code to use `EXPECT_NEAR` instead.

### CODE REVIEW

**Comment by davidwendt** [comment]
This may also fix #3847 failure on ppc64le.

File: cpp/custrings/tests/test_convert.cu [hunk 1]
```diff
@@ -134,8 +136,10 @@ TEST_F(TestConvert, ToFloat)
             double fval2 = expected[idx];
             if( std::isnan(fval1) )
                 EXPECT_TRUE( std::isnan(fval2) );
+            else if( std::isinf(fval1) )
+                EXPECT_TRUE( std::isinf(fval2) );
             else
-                EXPECT_EQ(fval1,fval2);
+                EXPECT_NEAR(fval1,fval2,1e-10);
```
Comment (vuule) [inline cpp/custrings/tests/test_convert.cu]: Have you tried using EXPECT_DOUBLE_EQ here?

Comment (davidwendt) [reply cpp/custrings/tests/test_convert.cu]: I tried that initially and it failed. Looks like I can get it to work if I update the expected value.

File: cpp/custrings/tests/test_convert.cu [hunk 2]
```diff
@@ -127,15 +129,17 @@ TEST_F(TestConvert, ToFloat)
         double expected[] = { 1234.0, 0,
             -876.0, 543.2, -0.12, 0.25, -0.002,
             0, nanval, 0, 123.0, 456.0, -178000.0,
-            -122.3364478212345, infval };
+            -122.33644782123469, infval };
```
Comment (trevorsm7) [inline cpp/custrings/tests/test_convert.cu]: Why not use the same `-122.33644782123456789` as above, and let the compiler truncate to float/double?

File: cpp/custrings/tests/test_convert.cu (hunk 1 above)
Comment (davidwendt) [reply cpp/custrings/tests/test_convert.cu]: So `EXPECT_DOUBLE_EQ` fails now on 10.1.
I tested `EXPECT_NEAR` on 10.1 and 10.2 so I think it should be ok.

File: cpp/custrings/tests/test_convert.cu (hunk 2 above)
Comment (davidwendt) [reply cpp/custrings/tests/test_convert.cu]: For some reason, this did not work with with `EXPECT_DOUBLE_EQ`.
I've changed it to `EXPECT_NEAR` now so I think that may be ok.

**Pull request merged**